src/IndexingMsg.chpl

```diff
@@ -145,25 +186,63 @@ module IndexingMsg
     /* arrayViewIntIndexAssignMsg "av[int_list]=value" response to __getitem__(int_list) where av is an ArrayView */
     proc arrayViewIntIndexAssignMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
         param pn = Reflection.getRoutineName();
-        var (pdaName, dtypeStr, dimProdName, coordsName, value) = payload.splitMsgToTuple(5);
+        var msgArgs = parseMessageArgs(payload, 5);
+        const pdaName = msgArgs.getValueOf("base");
+        const dimProdName = msgArgs.getValueOf("dim_prod");
+        const coordsName = msgArgs.getValueOf("coords");
+        const dtypeStr = msgArgs.getValueOf("dtype");
+        var value = msgArgs.getValueOf("value");
         imLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), "%s %s %s %s".format(cmd, pdaName, dimProdName, coordsName));
 
         var dimProd: borrowed GenSymEntry = getGenericTypedArrayEntry(dimProdName, st);
         var dimProdEntry = toSymEntry(dimProd, int);
         var coords: borrowed GenSymEntry = getGenericTypedArrayEntry(coordsName, st);
 
+        // map used to generate the "array" key for intIndexMsg
+        var arrayMap = new map(string, string);
+        arrayMap.add("key", "array");
+        arrayMap.add("val", pdaName);
+        arrayMap.add("objType", "PDARRAY");
+        arrayMap.add("dtype", "");
+
+        // map used to generate the "dtypestr" key for intIndexMsg
+        //TODO - idk if we need this because the json format passes the dtype string
+        var dtypeMap = new map(string, string);
+        dtypeMap.add("key", "dtype");
+        dtypeMap.add("val", dtypeStr);
+        dtypeMap.add("objType", "VALUE");
+        dtypeMap.add("dtype", "str");
+
+        // map used to generate the "value" key for intIndexMsg
+        var valMap = new map(string, string);
+        valMap.add("key", "value");
+        valMap.add("val", value);
+        valMap.add("objType", "VALUE");
+        valMap.add("dtype", dtypeStr);
+
```

Comment:
maybe we could make a helper function to generate these maps similar to how they are originally constructed client side? But i'm not sure if we call `*Msg` functions enough from the server to make it worthwhile

Reply:
I don't know that it is worth it right now. I am going to add an issue to look into this though because it could be useful if we expand these use cases.

Reply:
Added Issue #1731 to track this.